ocaml/xapi/xapi_vm.ml

```diff
@@ -221,6 +221,16 @@ let unpause ~__context ~vm =
 let set_xenstore_data ~__context ~self ~value =
   Xapi_xenops.set_xenstore_data ~__context ~self value
 
+(* CP-18860: check memory limits if using nested_virt on start *)
+let assert_memory_constraints ~__context ~vm platformdata =
```

Comment:
It seems to me that the memory checking functions called from `assert_can_boot_here` are for checking whether the given host has enough memory for the VM to start. This new constraint is independent from the host.

Also, if you add it to `assert_can_boot_here`, then I don't think there is an easy way to _not_ do it if `force=true`. We could add more parameters to it, but it does not seem the right place to me due to the above.